topostats/io.py

```diff
@@ -983,9 +1034,117 @@ def add_to_dict(self, image: np.ndarray, filename: str) -> None:
             "image_original": image,
             "image_flattened": None,
             "grain_masks": self.grain_masks,
+            "grain_trace_data": self.grain_trace_data,
         }
 
 
+def _hdf5_add_known_datatype(
+    open_hdf5_file: h5py.File, group_path: str, item: list | str | int | float | np.ndarray | Path | dict, key: str
```

Comment:
I did have this originally, but it failed, I can't remember why apparently tuples shouldn't be saved to HDF5. I'll try to find out why I removed it. It was an exception specially for it I think.

Reply:
Also a known datatype is just the datatypes that I have defined for the function to handle. Otherwise it will try to append it knowing that it could fail.

Reply:
Redundant, ruff no longer complains as having the function in one function